src/functionalTest/groovy/com/autonomousapps/jvm/projects/FeatureVariantInConsumerTestProject.groovy

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2024. Tony Robalik.
+// SPDX-License-Identifier: Apache-2.0
+package com.autonomousapps.jvm.projects
+
+import com.autonomousapps.AbstractProject
+import com.autonomousapps.kit.GradleProject
+import com.autonomousapps.kit.Source
+import com.autonomousapps.kit.SourceType
+import com.autonomousapps.kit.gradle.Feature
+import com.autonomousapps.kit.gradle.Java
+import com.autonomousapps.kit.gradle.Plugin
+import com.autonomousapps.model.Advice
+import com.autonomousapps.model.ProjectAdvice
+
+import static com.autonomousapps.AdviceHelper.*
+import static com.autonomousapps.kit.gradle.Dependency.project
+
+final class FeatureVariantInConsumerTestProject extends AbstractProject {
+
+  final GradleProject gradleProject
+
+  FeatureVariantInConsumerTestProject() {
+    this.gradleProject = build()
+  }
+
+  private GradleProject build() {
+    def builder = newGradleProjectBuilder()
+    builder.withSubproject('producer') { s ->
+      s.sources = producerSources
+      s.withBuildScript { bs ->
+        bs.plugins = javaLibrary
+      }
+    }
+    builder.withSubproject('consumer') { s ->
+      s.sources = consumerSources
+      s.withBuildScript { bs ->
+        bs.plugins = javaLibrary + Plugin.javaTestFixtures
+        bs.java = Java.ofFeatures(Feature.ofName('extra'))
+        bs.dependencies = [
+          project('api', ':producer'),
+          project('testFixturesImplementation', ':producer'),
+          project('extraImplementation', ':consumer')
```

Comment:
This dependency is saying that the `extra` source set on this project (the consumer project) has a dependency on the main source set, right?

Reply:
Yes. For `testFixtures` this is added by the test fixtures plugin.